Add ES Lint rules for `DynamicColorIOS()`and `ColorAndroid()`

tom-un opened the pull request:
## Summary

The [PlatformColor PR](https://github.com/facebook/react-native/pull/27908) added support for iOS and Android to express platform specific color values.   The primary method for an app to specify such colors is via the `PlatformColor()` method that takes string arguments.   The `PlatformColor` method returns an opaque Flow type enforcing that apps use the PlatformColor method instead of creating Objects from scratch -- doing so would make it harder to write static analysis tools around Color values in the future.   But in addition to `PlatformColor()`, iOS has a `DynamicColorIOS()` method that takes an Object.   The Flow type for this Object cannot be opaque, but we still want to enforce that app code doesn't pass variables instead of Object literals or that values in the Objects are variables.   To ensure `DynamicColorIOS()` can be statically analyzed this change adds an ESLint rule to enforce that `DynamicColorIOS()` takes an Object literal of a specific shape.   A `ColorAndroid()` was also introduced not for practical use but just to test having platform specific methods for more than one platform in the same app.   A second ESLint rule is created for `ColorAndroid` as well.

## Changelog

<!-- Help reviewers and the release process by writing your own changelog entry. See https://github.com/facebook/react-native/wiki/Changelog for an example. -->

[General] [Changed] - Add ES Lint rules for `DynamicColorIOS()`and `ColorAndroid()`

## Test Plan

`yarn lint` passes.

TheSavior commented:
Thanks for this! Can you look into writing tests for these lint rules? Check out https://github.com/facebook/react-native/blob/master/packages/eslint-plugin-react-native-community/__tests__/error-subclass-name-test.js for an example.

Lint rules are one of the things we push for really good tests for since the infra is so good for it.

tom-un commented:
> Thanks for this! Can you look into writing tests for these lint rules? Check out https://github.com/facebook/react-native/blob/master/packages/eslint-plugin-react-native-community/__tests__/error-subclass-name-test.js for an example.
> 
> Lint rules are one of the things we push for really good tests for since the infra is so good for it.
> 
> Also, do you think it makes sense for this to all be one rule vs a different rule for each function? I can't imagine people would want to turn them on or off separately.
> 
> I also think we need a similar rule for `PlatformColor` itself, making sure people pass in a string and not a variable. `PlatformColor('secondaryLabelColor ')` not `PlatformColor(myColor)`.

Done: added tests, merged into a single rule file, added `PlatformColor` rules.
